internal/tasks/subscriptionmanager.go

```diff
@@ -0,0 +1,259 @@
+			return ctx.Err()
+		case pairs := <-sm.watchCh:
+			logger.Ctx(ctx).Infow("received data from node bindings channel", "pairs", pairs)
+
+			// pairs can be be nil if it returns because of closing of channel
+			if pairs == nil {
+				return nil
+			}
+
+			err := sm.handleNodeBindingUpdates(ctx, pairs)
+			if err != nil {
+				logger.Ctx(ctx).Infow("error processing nodebinding updates", "error", err)
+				return err
+			}
+		}
+	}
+}
+
+func (sm *SubscriptionManager) handleNodeBindingUpdates(ctx context.Context, newBindingPairs []registry.Pair) error {
+	oldBindings := sm.nodebindingCache
+	var newBindings []*nodebinding.Model
+
+	for _, pair := range newBindingPairs {
+		nb := nodebinding.Model{}
+		err := json.Unmarshal(pair.Value, &nb)
+		if err != nil {
+			return err
+		}
+		newBindings = append(newBindings, &nb)
+	}
+
+	for _, old := range oldBindings {
+		oldKey := old.Key()
+		found := false
+		for _, newBinding := range newBindings {
+			if oldKey == newBinding.Key() {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			logger.Ctx(ctx).Infow("binding removed", "key", oldKey)
+			if handler, ok := sm.pushHandlers.Load(oldKey); ok && handler != nil {
+				logger.Ctx(ctx).Infow("handler found, calling stop", "key", oldKey)
+				go func(ctx context.Context) {
+					err := handler.(*stream.PushStream).Stop()
+					if err == nil {
+						logger.Ctx(ctx).Infow("handler stopped", "key", oldKey)
+					}
+				}(ctx)
+				sm.pushHandlers.Delete(oldKey)
+			}
+		}
+	}
+
+	for _, newBinding := range newBindings {
+		found := false
+		for _, old := range oldBindings {
+			if old.Key() == newBinding.Key() {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			logger.Ctx(ctx).Infow("binding added", "key", newBinding.Key())
+			handler := stream.NewPushStream(ctx, newBinding.ID, newBinding.SubscriptionID, sm.subscriptionCore, sm.subscriber, sm.httpConfig)
+
+			// run the stream in a separate go routine, this go routine is not part of the worker error group
+			// as the worker should continue to run if a single subscription stream exists with error
+			go func(ctx context.Context) {
+				err := handler.Start()
+				if err != nil {
+					logger.Ctx(ctx).Errorw("[worker]: push stream handler exited",
+						"subscription", newBinding.SubscriptionID,
+						"error", err.Error())
+				}
+			}(ctx)
+
+			// store the handler
+			sm.pushHandlers.Store(newBinding.Key(), handler)
+		}
+	}
+
+	sm.nodebindingCache = newBindings
+	return nil
+}
+
+func (sm *SubscriptionManager) stopPushHandlers(ctx context.Context) {
+	// stop all push stream handlers
+	logger.Ctx(ctx).Infow("metro stop: stopping all push handlers")
+	wg := sync.WaitGroup{}
+	sm.pushHandlers.Range(func(_, handler interface{}) bool {
+		wg.Add(1)
+		ps := handler.(*stream.PushStream)
+		go func(ps *stream.PushStream, wg *sync.WaitGroup) {
+			defer wg.Done()
+
+			err := ps.Stop()
+			if err != nil {
+				logger.Ctx(ctx).Infow("error stopping stream handler", "error", err)
```

Comment:
```suggestion
				logger.Ctx(ctx).Errorw("error stopping stream handler", "error", err)
```
check other places as well